src/diff/children.js

```diff
@@ -32,6 +32,26 @@ export function diffChildren(parentDom, newParentVNode, oldParentVNode, context,
 
 	let oldChildrenLength = oldChildren.length;
 
+	if (oldDom == EMPTY_OBJ) {
```

Comment:
Only in very specific places should this logic be invoked (top level `render` and `diffElementNodes`). I'm using `EMPTY_OBJ` to signal when `diffChildren` is invoked in these situations. I can't use `null` for this purpose, because `null` is a valid value for `oldDom` which can mean to skip to this logic (e.g. if mounting a new tree in which the old DOM should be ignored (usually for Fragments).

I tried using `=== undefined`  and `== EMPTY_ARR` but both ended up being a couple bytes larger.

Reply:
I'll add this as a comment in the code

Reply:
That's really clever, I was way too focused on finding any differences in the `vnodes` themselves rather than just checking function arguments :+1: